File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    
    map = {}
    
    for c_name, c_type in clothes:
        if c_type not in map:
            map[c_type] = 2
        else:
            map[c_type] = map[c_type] + 1
    
    res = 1
    
    for v in map.values():
        res *= v
    
    return res - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        
        HashMap<String, Integer> map = new HashMap<>();
        
        for(String[] c : clothes) {            
            if (map.containsKey(c[1])) {
                map.put(c[1], map.get(c[1]) + 1);
            } else {                
                map.put(c[1], 2);
            }
        }
        
        int answer = 1;
        
        for(HashMap.Entry<String, Integer> val : map.entrySet()) {
            answer *= val.getValue();
        }
        
        
        return answer - 1;
    }
}